llvm/lib/CodeGen/TargetRegisterInfo.cpp

```diff
@@ -674,6 +674,50 @@ TargetRegisterInfo::prependOffsetExpression(const DIExpression *Expr,
                                       PrependFlags & DIExpression::EntryValue);
 }
 
+unsigned TargetRegisterInfo::getRegPressureSetLimit(const MachineFunction &MF,
+                                                    unsigned Idx) const {
+  const TargetRegisterClass *RC = nullptr;
+  unsigned NumRCUnits = 0;
+  for (const TargetRegisterClass *C : regclasses()) {
+    const int *PSetID = getRegClassPressureSets(C);
+    for (; *PSetID != -1; ++PSetID) {
+      if ((unsigned)*PSetID == Idx)
+        break;
+    }
+    if (*PSetID == -1)
+      continue;
+
+    // Found a register class that counts against this pressure set.
+    // For efficiency, only compute the set order for the largest set.
+    unsigned NUnits = getRegClassWeight(C).WeightLimit;
+    if (!RC || NUnits > NumRCUnits) {
+      RC = C;
+      NumRCUnits = NUnits;
+    }
+  }
+  assert(RC && "Failed to find register class");
+
+  unsigned NReserved = 0;
+  const BitVector Reserved = MF.getRegInfo().getReservedRegs();
```

Comment:
This is the same as the following:
```suggestion
  const BitVector Reserved = getReservedRegs();
```

Reply:
It is different I think, `getReservedRegs()` will recalculate the reserved set, but `MF.getRegInfo().getReservedRegs()` will the cached one.

Reply:
Sorry, yeah, I got confused between `TargetRegisterInfo` and `MachineRegisterInfo`.

Reply:
We should rename the TRI version to something else. It's nearly always the wrong thing to use